frontend/src/views/WheresMyGene/components/HeatMap/components/Chart/components/Chart/index.tsx

```diff
@@ -0,0 +1,110 @@
+import { ECharts, ElementEvent, init } from "echarts";
+import {
+  memo,
+  useEffect,
+  useRef,
+  useState,
+  forwardRef,
+  ForwardedRef,
+  HTMLAttributes,
+} from "react";
+
+import { EMPTY_OBJECT } from "src/common/constants/utils";
+
+import { ChartContainer } from "./style";
+import { useUpdateChart } from "./hooks/useUpdateChart";
+import { CreateChartOptionsProps } from "./hooks/utils";
+
+export interface Props<Data, XAxisData, YAxisData>
+  extends HTMLAttributes<HTMLDivElement>,
+    CreateChartOptionsProps<Data, XAxisData, YAxisData> {
+  echartsRendererMode?: "svg" | "canvas";
+  onChartMouseMove?: (event: ElementEvent, chart: ECharts) => void;
+}
+
+function Chart<Data, XAxisData, YAxisData>(
+  props: Props<Data, XAxisData, YAxisData>,
+  ref: ForwardedRef<HTMLDivElement>
+): JSX.Element {
+  const {
+    width,
+    height,
+    echartsRendererMode = "canvas",
+    onChartMouseMove,
+    xAxisData,
+    yAxisData,
+    data,
+    encode,
+    itemStyle,
+    symbolSize,
+    grid,
+    ...rest
+  } = props;
+
+  const [isChartInitialized, setIsChartInitialized] = useState(false);
+
+  const [chart, setChart] = useState<ECharts | null>(null);
+  const innerRef = useRef<HTMLDivElement | null>(null);
+
+  // Initialize charts
+  useEffect(() => {
+    const { current } = innerRef;
+
+    if (
+      !current ||
+      isChartInitialized ||
+      // (thuang): echart's `init()` will throw error if the container has 0 width or height
+      current?.getAttribute("height") === "0" ||
+      current?.getAttribute("width") === "0"
+    ) {
+      return;
```

Comment:
Should we throw an error here alongside?

Reply:
Ooh great idea! Will do

Reply:
Done! Added the following at the top of component:

```ts
if (!width || !height) {
  throw Error("Chart must have width and height > 0");
}
```